presto-orc/src/test/java/com/facebook/presto/orc/metadata/statistics/TestStringStatisticsBuilder.java

```diff
@@ -76,4 +82,53 @@ public void testMinMaxValues()
 
         assertMinMaxValues(HIGH_BOTTOM_VALUE, HIGH_TOP_VALUE);
     }
+
+    @Test
+    public void testSum()
+    {
+        StringStatisticsBuilder stringStatisticsBuilder = new StringStatisticsBuilder();
+        for (Slice value : ImmutableList.of(EMPTY_SLICE, LOW_BOTTOM_VALUE, LOW_TOP_VALUE)) {
+            stringStatisticsBuilder.addValue(value);
+        }
+        assertStringStatistics(stringStatisticsBuilder.buildColumnStatistics(), 3, EMPTY_SLICE.length() + LOW_BOTTOM_VALUE.length() + LOW_TOP_VALUE.length());
+    }
+
+    @Test
+    public void testMerge()
+    {
+        List<ColumnStatistics> statisticsList = new ArrayList<>();
+
+        StringStatisticsBuilder statisticsBuilder = new StringStatisticsBuilder();
+        statisticsList.add(statisticsBuilder.buildColumnStatistics());
+        assertMergedStringStatistics(statisticsList, 0, 0);
+
+        statisticsBuilder.addValue(EMPTY_SLICE);
+        statisticsList.add(statisticsBuilder.buildColumnStatistics());
+        assertMergedStringStatistics(statisticsList, 1, 0);
+
+        statisticsBuilder.addValue(LOW_BOTTOM_VALUE);
+        statisticsList.add(statisticsBuilder.buildColumnStatistics());
+        assertMergedStringStatistics(statisticsList, 3, LOW_BOTTOM_VALUE.length());
+
+        statisticsBuilder.addValue(LOW_TOP_VALUE);
+        statisticsList.add(statisticsBuilder.buildColumnStatistics());
+        assertMergedStringStatistics(statisticsList, 6, LOW_BOTTOM_VALUE.length() * 2 + LOW_TOP_VALUE.length());
+    }
+
+    private void assertMergedStringStatistics(List<ColumnStatistics> statisticsList, int expectedNumberOfValues, long expectedSum)
+    {
+        assertStringStatistics(mergeColumnStatistics(statisticsList), expectedNumberOfValues, expectedSum);
+
+        assertNoColumnStatistics(mergeColumnStatistics(insertEmptyColumnStatisticsAt(statisticsList, 0, 10)), expectedNumberOfValues + 10);
+        assertNoColumnStatistics(mergeColumnStatistics(insertEmptyColumnStatisticsAt(statisticsList, statisticsList.size(), 10)), expectedNumberOfValues + 10);
+        assertNoColumnStatistics(mergeColumnStatistics(insertEmptyColumnStatisticsAt(statisticsList, statisticsList.size() / 2, 10)), expectedNumberOfValues + 10);
+    }
+
+    private void assertStringStatistics(ColumnStatistics columnStatistics, int expectedNumberOfValues, long expectedSum)
+    {
+        if (expectedNumberOfValues > 0) {
```

Comment:
verify number of values is zero when expectedNumberOfValues is zero?  Also if length is present it should be zero also?